libsql/internal/hrana/batch_result.go

```diff
@@ -1,6 +1,49 @@
 package hrana
 
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+)
+
 type BatchResult struct {
-	StepResults []*StmtResult `json:"step_results"`
-	StepErrors  []*Error      `json:"step_errors"`
+	StepResults      []*StmtResult `json:"step_results"`
+	StepErrors       []*Error      `json:"step_errors"`
+	ReplicationIndex *uint64       `json:"replication_index"`
+}
+
+func (b *BatchResult) UnmarshalJSON(data []byte) error {
+	type Alias BatchResult
+	aux := &struct {
+		ReplicationIndex interface{} `json:"replication_index,omitempty"`
+		*Alias
+	}{
+		Alias: (*Alias)(b),
+	}
+
+	if err := json.Unmarshal(data, &aux); err != nil {
+		return err
+	}
+
+	if aux.ReplicationIndex == nil {
+		return nil
+	}
+
+	switch v := aux.ReplicationIndex.(type) {
+	case float64:
+		repIndex := uint64(v)
+		b.ReplicationIndex = &repIndex
```

Comment:
It took me some time to realize that it's go and there is a GC 😄 I was just horrified at seeing you return a reference from the stack

Reply:
wait is that actually on the heap? I'm confused

Reply:
it's on the heap - chill :)